jerry-core/parser/js/js-parser-expr.c

```diff
@@ -2765,6 +2765,18 @@ parser_pattern_process_assignment (parser_context_t *context_p, /**< context */
            CBC_EXT_REST_INITIALIZER -> CBC_EXT_ITERATOR_STEP_2 */
         uint16_t stack_difference = (uint16_t) (context_p->stack_depth - stack_depth_before);
 
+        /* During forming assignment CBC_PUSH_PROP_LITERAL is going to be transformed to
+           - PUSH LITERAL + CBC_ASSIGN therefore the stack difference should be increased
+           Example: [a[b][c]]
+           CBC_EXT_ITERATOR_STEP -> CBC_EXT_ITERATOR_STEP_3
+           CBC_EXT_REST_INITIALIZER -> CBC_EXT_ITERATOR_STEP_3 */
+        if (stack_difference == 1
+            && context_p->last_cbc_opcode == CBC_PUSH_PROP_LITERAL
+            && context_p->last_cbc.literal_type == LEXER_IDENT_LITERAL)
+        {
+          stack_difference++;
+        }
+
         if (context_p->last_cbc_opcode == CBC_PUSH_PROP_LITERAL_LITERAL)
```

Comment:
Would it make sense to make this `if` to `else if`? If I read this correctly: if the first check is true, then this check is not needed.

Reply:
I've reworked the PR, now this check is no longer needed.